### PR TITLE
Implemented unit tests on header component

### LINKED ISSUE #5: setup Jest and React testing library for unit testing
- [ ] Install jest, React testing library
- [ ] write unit tests for Header.tsx component

Help URL: https://nextjs.org/docs/pages/building-your-application/optimizing/testing

### PR DESCRIPTION
**Pull request by henriquedevelops**
Fixes #5 

I implemented tests for rendering the header title, rendering the GitHub icon and switching the moon/sun icons when the theme changes.

Besides that, the initial state of the "theme" variable was reversed: when there is no system preference the theme is initially set to light, but the code was written as if it was initially set to dark. The moon icon was setting the theme to light and the sun icon was setting it to dark, which doesn't make sense, right? So I also fixed this by reversing all darks to light and all lights to dark.

### CODE REVIEW

File: app/components/Header.test.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import '@testing-library/jest-dom'
+import { fireEvent, render, screen, waitFor } from '@testing-library/react'
```
Comment (kirandash) [inline app/components/Header.test.tsx:2]: although fireEvent is not incorrect. userEvent is a better way of testing user events.

you can see how it's implemented on another project of mine at: 

https://github.com/bgwebagency/redesigner.io/blob/f5794c29d77ae7b5384aebc6f02d248c4fce7edf/app/components/Footer/__test__/Footer.test.tsx#L4

Read more at: https://testing-library.com/docs/user-event/intro/#differences-from-fireevent

**Review by kirandash — CHANGES_REQUESTED** [review]

File: app/components/Header.tsx [hunk 2]
```diff
@@ -22,28 +22,30 @@ const Header = () => {
...
-                setTheme("light");
-                document.body.classList.toggle("dark");
+                setTheme('dark')
+                document.body.classList.toggle('dark')
               }}
             />
           ) : (
             <BsSunFill
               size={32}
+              aria-label="Moon"
```
[the remark this answers is not in the document]
Comment (henriquebuzon) [reply app/components/Header.tsx]: Did it!

File: app/components/Header.test.tsx (hunk 1 above)
Comment (henriquebuzon) [reply app/components/Header.test.tsx]: Did it!

Comment (henriquebuzon) [reply app/components/Header.test.tsx]: Done!

File: app/components/Header.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (henriquebuzon) [reply app/components/Header.tsx]: Done!

**Review by kirandash — APPROVED** [review]

**Pull request merged**